### PR TITLE
Add the primary key reading from the GenericRow

### PR DESCRIPTION
**Pull request by yupeng9**
## Description
Part of a series of PRs for #4261
Check this [doc](https://docs.google.com/document/d/1qljEMndPMxbbKtjlVn9mn2toz7Qrk0TGQsHLfI--7h8/edit#heading=h.lsfmyoyyxtgt) out for the new design

Related to https://github.com/apache/incubator-pinot/pull/6096

### CODE REVIEW

File: pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/GenericRow.java [hunk 1]
```diff
@@ -123,6 +124,14 @@ public void putValue(String fieldName, @Nullable Object value) {
     _fieldToValueMap.put(fieldName, value);
   }
 
+  public PrimaryKey getPrimaryKey(List<String> primaryKeyColumns) {
+    Object[] fields = new Object[primaryKeyColumns.size()];
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/GenericRow.java:128]: Cache `primaryKeyColumns.size()` into a local variable

Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/GenericRow.java:128]: Suggest renaming it to `values`

File: pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/PrimaryKey.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+package org.apache.pinot.spi.data.readers;
+
+import java.util.Arrays;
+
+
+/**
+ * The primary key of a record.
+ */
+public class PrimaryKey {
+  private final Object[] _fields;
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/PrimaryKey.java:28]: Suggest renaming it to `_values`

File: pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/PrimaryKey.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.data.readers;
+
+import java.util.Arrays;
+
+
+/**
+ * The primary key of a record.
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/PrimaryKey.java:25]: Add some javadoc stating that the values should be SV (MV values won't work for equals and hashcode)

**Review by Jackie-Jiang — APPROVED** [review]
LGTM with minor comments

File: pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/PrimaryKey.java (hunk 3 above)
Comment (yupeng9) [reply pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/PrimaryKey.java:25]: Yes. Table config shall validate this.

**Comment by yupeng9** [comment]
@Jackie-Jiang thanks for the review. Comments addressed.

File: pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/GenericRow.java [hunk 4]
```diff
@@ -123,6 +124,15 @@ public void putValue(String fieldName, @Nullable Object value) {
     _fieldToValueMap.put(fieldName, value);
   }
 
+  public PrimaryKey getPrimaryKey(List<String> primaryKeyColumns) {
+    Object[] values = new Object[primaryKeyColumns.size()];
+    int size = primaryKeyColumns.size();
+    for (int i = 0; i < size; i++) {
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/GenericRow.java:130]: (nit)
```suggestion
    int numPrimaryKeyColumns = primaryKeyColumns.size();
    Object[] values = new Object[numPrimaryKeyColumns];
    for (int i = 0; i < numPrimaryKeyColumns; i++) {
```

File: pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/PrimaryKey.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+
+/**
+ * The primary key of a record. Note that the value used in the primary key must be single-value.
+ */
+public class PrimaryKey {
+  private final Object[] _values;
+
+  public PrimaryKey(Object[] fields) {
+    _values = fields;
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/PrimaryKey.java:31]: (nit)
```suggestion
  public PrimaryKey(Object[] values) {
    _values = values;
```

**Review by Jackie-Jiang — COMMENTED** [review]
Minor comments

**Comment by yupeng9** [comment]
@Jackie-Jiang done. Thanks for taking another look

**Pull request merged**